modin/experimental/engines/omnisci_on_native/io.py

```diff
@@ -245,82 +244,14 @@ def read_csv(
             )
             if not use_modin_impl:
                 raise ArrowEngineException(error_message)
-
-            # TODO: potentially do this only in a strict compatibility mode as this may
-            # really harm performance for wide frames
-            # if STRICT_COMPATIBILITY:
```

Comment:
If there is an issue about compatibility mode, please add that info to it.
If there is no such issue make one and add that info to it :)

Reply:
I've already created one (#3485) and also pointed there the reverted changes. Everywhere in this PR where the workaround of this bug is visible, I'm already referring to the newly created issue [[1]](https://github.com/modin-project/modin/pull/3308/files#diff-78cc8a1e065a59145ed94d6f661e112d4f4226826c6a2ba7f854aad3c54324e4R55), [[2]](https://github.com/modin-project/modin/pull/3308/files#diff-c5eeb2d8c13631b353db882e70a405b4df07007260f029fa4b2375ea7a01350aR1112)